ChaosMod/Effects/db/Player/PlayerGoPro.cpp

```diff
@@ -0,0 +1,99 @@
+/*
+    Effect by Chanka
+*/
+
+#include <stdafx.h>
+
+Cam goPro;
+Entity boneProxy;
+int state = 0;
+
+static bool SafetyCheck()
+{
+	//Removes camera on player switch, death, and potential non-existence
+	if (IS_PLAYER_SWITCH_IN_PROGRESS() || IS_PLAYER_DEAD(PLAYER_ID()) || !ENTITY::DOES_ENTITY_EXIST(PLAYER_PED_ID())
+	    || !ENTITY::IS_ENTITY_VISIBLE(PLAYER_PED_ID()))
+	{
+		return true;
+	}
+	return false;
+}
+
+static void OnStart()
+{
+	//Not sure of a direct way to obtain bone rotation. Attaching a prop is a workaround.
+	Vector3 boneCoords = GET_PED_BONE_COORDS(PLAYER_PED_ID(), 0x322c, 0, 0, 0);
+	goPro = CREATE_CAM_WITH_PARAMS("DEFAULT_SCRIPTED_CAMERA", boneCoords.x, boneCoords.y, boneCoords.z, 0, 0, 0, 60, false, 2);
+	boneProxy = CREATE_OBJECT("p_cs_leaf_s"_hash, boneCoords.x, boneCoords.y, boneCoords.z, false, false, true);
+	SET_ENTITY_ALPHA(boneProxy, 0, false);
+	ATTACH_ENTITY_TO_ENTITY(boneProxy, PLAYER_PED_ID(), GET_PED_BONE_INDEX(PLAYER_PED_ID(), 0x322c), 0.1, 0, 0, 0, 90,
```

Comment:
I would cache the player ped in a variable to reduce the amount of times you have the call for it.